deepchecks/tabular/checks/data_integrity/feature_feature_correlation.py

```diff
@@ -0,0 +1,96 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""module contains the Feature-Feature Correlation check."""
+
+import pandas as pd
+import plotly.express as px
+
+from deepchecks.core import CheckResult, ConditionCategory, ConditionResult
+from deepchecks.tabular import Context, SingleDatasetCheck
+from deepchecks.utils.correlation_methods import correlation_ratio, symmetric_theil_u_correlation
+from deepchecks.utils.dataframes import generalized_corrwith
+
+__all__ = ['FeatureFeatureCorrelation']
+
+
+class FeatureFeatureCorrelation(SingleDatasetCheck):
+    """
+    Checks for pairwise correlation between the features.
+
+    Extremely correlated pairs could indicate redundancy and even duplication.
+    """
+
+    def __init__(self,
+                 **kwargs):
+        super().__init__()
+
+    def run_logic(self, context: Context, dataset_type: str = 'train') -> CheckResult:
+        """
+        Run Check.
+
+        Returns
+        -------
+        CheckResult
+            A DataFrame of the pairwise correlations between the features.
+        """
+        if dataset_type == 'train':
+            dataset = context.train
+        else:
+            dataset = context.test
+
+        dataset.assert_features()
+        num_features = dataset.numerical_features
+        cat_features = dataset.cat_features
+
+        encoded_cat_data = dataset.data.loc[:, cat_features].apply(lambda x: pd.factorize(x)[0])
+
+        all_features = num_features + cat_features
+        full_df = pd.DataFrame(index=all_features, columns=all_features)
+
+        # Numerical-numerical correlations
+        if num_features:
+            full_df.loc[num_features, num_features] = dataset.data.loc[:, num_features].corr(method='spearman')
+
+        # Categorical-categorical correlations
+        if cat_features:
+            full_df.loc[cat_features, cat_features] = encoded_cat_data.corr(method=symmetric_theil_u_correlation)
+
+        # Numerical-categorical correlations
+        if num_features and cat_features:
+            num_cat_corr = generalized_corrwith(dataset.data.loc[:, num_features], encoded_cat_data,
+                                                method=correlation_ratio)
+            full_df.loc[num_features, cat_features] = num_cat_corr
+            full_df.loc[cat_features, num_features] = num_cat_corr.transpose()
+
+        # Display
+        fig = px.imshow(full_df)
```

Comment:
What happens if we have 1000 features? will the plot be readable?
Maybe we should limit to the top X higher correlations?

Reply:
I think that the highest are not necessarily the most interesting, and the user should just take this into account and run the check on the partial dataset that they like. Showing only the top could create visual perception bias because the color scheme will be spread on a narrow scale.

Reply:
@ItayGabbay I've implemented shot_top_n_columns